test/abstract_client.js

```diff
@@ -780,6 +781,73 @@ module.exports = function (server, config) {
     it('should publish 10 QoS 2 and receive them only when `handleMessage` finishes', function (done) {
       testQosHandleMessage(2, done)
     })
+
+    it('should not send a `puback` if the execution of `handleMessage` fails for messages with QoS `1`', function (done) {
+      var client = connect()
+
+      client.handleMessage = function (packet, callback) {
+        callback(new Error('Error thrown by the application'))
+      }
+
+      client._sendPacket = sinon.spy()
+
+      client._handlePublish({
```

Comment:
@mcollina The only possible way I could think of for writing a test case (without using setTimeouts) against this particular functionality is by directly manipulating the '_handlePublish' private method, which is sub-optimal for obvious reasons. I'd appreciate if you could shed some light as to if there's any other better way to handle this.

Reply:
this is ok.

Reply:
Thanks for the feedback.